Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,73 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        private Stream Source { get; set; }
         public DataLoader(Stream source)
         {
+            Source = source;
```

Comment:
`source` should be null checked
```suggestion
            Source = source ?? throw new ArgumentNullException(nameof(source));
```